ssaHAL/ssa/interfaces/runtime.py

```
@@ -0,0 +1,120 @@
+
+class SSARuntime(SSAConnector, AffordanceHandler):
+    """Base class defining the SSARuntime interface."""
+    def __init__(self, config):
+        """Initialize SSARuntime with a required configuration dictionary."""
+        self.config = config
+
+        default_config_template = {
+            "tm": {"name": str, "version": {"model": str, "instance": str}}
+        }
+
+        extra_config_template.update(default_config_template)
```

Ruff (0.8.2)

Undefined name `extra_config_template`

(F821)